src/AppInstallerCLICore/PackageCollection.cpp

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+
+#include "PackageCollection.h"
+#include "AppInstallerRuntime.h"
+
+#include <algorithm>
+#include <ostream>
+
+namespace AppInstaller::CLI
+{
+    using namespace AppInstaller::CLI::PackagesJson;
+    using namespace AppInstaller::Repository;
+
+    namespace
+    {
+        // Gets a property of a JSON object by its name.
+        template<Json::ValueType T>
+        Json::Value& GetJsonProperty(Json::Value& node, const std::string& propertyName)
+        {
+            if (!node.isMember(propertyName))
+            {
+                node[propertyName] = Json::Value{ T };
+            }
+
+            THROW_HR_IF(E_NOT_VALID_STATE, node[propertyName].type() != T);
+            return node[propertyName];
+        }
+
+        // Sets a property of a JSON object to a string if it is not empty
+        void SetJsonProperty(Json::Value& node, const std::string& propertyName, const std::string& value)
+        {
+            if (!value.empty())
+            {
+                node[propertyName] = value;
+            }
+            else if (node.isMember(propertyName))
+            {
+                node.removeMember(propertyName);
+            }
+        }
+
+        // Reads the description of a package from a Package node in the JSON.
+        PackageRequest ParsePackageNode(const Json::Value& packageNode)
+        {
+            std::string id = packageNode[PACKAGE_ID_PROPERTY].asString();
```

Comment:
I haven't done it, but it should be checked by validating the schema. But since you say we may not have an Id, I'll probably have to check here anyway.